deploy/docker/Dockerfile-alpine

```diff
@@ -11,20 +11,36 @@ RUN make build_with_edgex
 
 FROM alpine:3.12
 
+# Set environment vars
+ENV MAINTAINER="emqx.io" \
+    KUIPER_HOME="/kuiper" \
+    KUIPER__BASIC__CONSOLELOG=true
+
+# These vars are not persisted in the final image layer
+ARG KUIPER_USER="kuiper"
+ARG KUIPER_USER_ID="1001"
+
+# (root) Add packages and "kuiper" user
+RUN apk add sed libzmq && \
+    adduser -DH -s /sbin/nologin -u ${KUIPER_USER_ID} ${KUIPER_USER}
+
 COPY ./deploy/docker/docker-entrypoint.sh /usr/bin/docker-entrypoint.sh
 COPY --from=builder /go/kuiper/kuiper_conf_util /usr/bin/kuiper_conf_util
 COPY --from=builder /go/kuiper/_build/kuiper-* /kuiper/
 
-RUN apk add sed libzmq
+WORKDIR ${KUIPER_HOME}
 
-WORKDIR /kuiper
+VOLUME ["${KUIPER_HOME}/etc", "${KUIPER_HOME}/data", "${KUIPER_HOME}/plugins", "${KUIPER_HOME}/log"]
+EXPOSE 9081 20498
 
-ENV MAINTAINER="emqx.io"
-ENV KUIPER_HOME /kuiper
-ENV KUIPER__BASIC__CONSOLELOG true
+# Set appropriate ownership to allow binary full access to KUIPER_HOME dir
+RUN chown -Rh ${KUIPER_USER}:${KUIPER_USER} ${KUIPER_HOME}
 
-VOLUME ["/kuiper/etc", "/kuiper/data", "/kuiper/plugins", "/kuiper/log"]
-EXPOSE 9081 20498
+# Create taos db directory and set kuiper to owner
+RUN mkdir -p /usr/local/taos && chown -Rh ${KUIPER_USER}:${KUIPER_USER} /usr/local/taos
+
```

Comment:
I think `adduser` and `chown` can be combined in the same `RUN`, which makes it easier to read, don't you think?

Reply:
No, they are separated because of build timing - "root" has to create the user and perform build processes, and then assign ownership after the build processes are complete.

Reply:
Updated - push coming shortly.